backend/x86_binary_emitter.ml

```diff
@@ -438,7 +436,7 @@ let emit_mod_rm_reg b rex opcodes rm reg =
       emit_rex b (rex lor rex_of_reg16 reg16 lor rexr_reg reg lor rexb_rm rm);
       buf_opcodes b opcodes;
       buf_int8 b (mod_rm_reg 0b11 rm reg)
-  | Regf rm ->
+  | Regf (XMM _ as rm) ->
```

Comment:
Why do we need to match on XMM here? I don't think it will help when other constructors are added to `regf` .